### PR TITLE
Update TAGAN-axons

### PR DESCRIPTION
**Pull request by bioimageiobot**
This is an automatic PR created by the @bioimageiobot regarding changes to the resource item 10.5281/zenodo.7872357.
The following version(s) will be added:
  - [7872358 (version 1)](https://www.doi.org/10.5281/zenodo.7872358)

Please review the changes and make sure the new item or version(s) pass the following check list:
- [ ] Passed the BioImage.IO CI tests: static (and dynamic) validations
- [ ] The meta information for the RDF item is complete
  - [ ] The tags are complete and describe the model
  - [ ] Naming is intuitive and descriptive, example:  Multi-Organ Nucleus Segmentation (StarDist 2D)
  - [ ] Authors are provided
  - [ ] Documentation is complete
    * For models, include an overview, describe how the model is trained, what is the training data, how to use the model, how to validate the results and list the references. TODO: Model documentation template.
- [ ] Approved by at least one of the BioImage.IO admin team member.

Maintainers: @cbouchard22

Note: If you updated or re-uploaded another version for the current item on Zenodo, this PR won't be changed automatically. To proceed, you can do the following:
 1. Block this version, but keep looking for future versions: Edit the current resource.yaml and keep the top-level `status` field as `accepted`, but change the `status` under the current version to `blocked`.
 2. Accept this version and keep looking for future versions: Merge this PR for now.
 3. Keep proposed version(s) (and this resource in general if it is new) as pending: Close this PR without merging.

 Then wait for the CI on the main branch to complete. It should detect the new version(s) and create another PR for the new version(s).

Previous PRs of this resource: none

### CODE REVIEW

**Comment by FynnBe** [comment]
Hello @cbouchard22, thank you for contributing this model!

The error `Could not run 'aten::empty_strided' with arguments from the 'CUDA' backend....` looks like you traced a model on GPU and CI fails when attempting inference on CPU.
This usually works, but depending on model architectures it can fail. 
a similar problem was caused by details of the architecture [here](https://discuss.huggingface.co/t/pytorch-trace-on-cpu-and-use-on-gpu/181/5 ).
In general PyTorch seems to suggest to trace twice for CPU/GPU:

```python
cpu_model = gpu_model.cpu()
sample_input_cpu = sample_input_gpu.cpu()
traced_cpu = torch.jit.trace(cpu_model, sample_input_cpu)
torch.jit.save(traced_cpu, "cpu.pt")
traced_gpu = torch.jit.trace(gpu_model, sample_input_gpu)
torch.jit.save(traced_gpu, "gpu.pt")
```
--- https://pytorch.org/docs/stable/jit.html#frequently-asked-questions

We should investigate this further and then maybe discuss how one can add two traced models (or more for different input sizes?) in the model RDF

Due to issues with tracing like this and other limitations of tracing, it would be amazing if you, @cbouchard22, could include [`pytorch_state_dict` weights](https://github.com/bioimage-io/spec-bioimage-io/blob/gh-pages/weight_formats_spec_latest.md#pytorch_state_dict) as well
With the pytorch_state_dict weights we can also take a look at what might cause the above error with the traced model.

cc @k-dominik @constantinpape @oeway

**Comment by cbouchard22** [comment]
Hello @FynnBe! Thank you for the and quick feedback. I've uploaded more files to Zenodo that should hopefully help troubleshoot the problems. You should now have access to the pytorch_state_dict weights (TAGAN_Axons_G.pth), the cpu version of the model (cpu.pt) and the gpu version (gpu.pt). I hope this helps resolve the problems.

Note that during the verification performed with the model_creation notebook, the model could not pass the last test "reproduce test outputs from test inputs", because there is stochasticity in the model (dropout layers) that makes the output slightly different for each inference pass. I'm not sure if there is a way to freeze the dropout layers in the traced model, so that this test can be successful?

**Comment by FynnBe** [comment]
> I'm not sure if there is a way to freeze the dropout layers in the traced model, so that this test can be successful?

Dropout layers are supposed to be active only during training. calling `model.eval()` (done by bioimageio.core) will skip them for inference.
If that does not happen, the model might not have been initialized correctly:
1. the model needs to inherite from [torch.nn.Module](https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module))
2. you need to call super().__init__() before adding other torch.nn.Module attributes (aka your layers)

For the pytorch_state_dict weights to make sense we'll need a model.py file as well that has the model source code as documented [here](https://github.com/bioimage-io/spec-bioimage-io/blob/gh-pages/weight_formats_spec_latest.md#pytorch_state_dict)

**Comment by FynnBe** [comment]
The `torchscript` validation fails. It looks related to tracing on GPU and not on CPU, which our test environment requires. 
To fix it please trace the model on cpu or script it in order to be able to execute it on CPU.
